Components/GoogleAnalyticsOptIn/script.js

```diff
@@ -0,0 +1,152 @@
+    this.$declineButton.on('click', this.declineTracking)
+    this.$optOutLink.on('click', this.showModal)
+  }
+
+  connectedCallback () {
+    this.checkOptState()
+  }
+
+  addGTAGScript () {
+    const scriptUrl = `https://www.googletagmanager.com/gtag/js?id=${this.gaId}`
+    this.GTAGscriptElem = document.createElement('script')
+    this.GTAGscriptElem.async = true
+    this.GTAGscriptElem.type = 'text/javascript'
+    this.GTAGscriptElem.src = scriptUrl
+    document.head.appendChild(this.GTAGscriptElem)
+  }
+
+  addGTAGFunction () {
+    const scriptContent = `window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+    gtag('js', new Date());
+    gtag('config', '${this.gaId}', { 'anonymize_ip': true });`
+    this.GTAGscriptElemFunction = document.createElement('script')
+    this.GTAGscriptElemFunction.type = 'text/javascript'
+    this.GTAGscriptElemFunction.text = scriptContent
+    document.head.appendChild(this.GTAGscriptElemFunction)
+  }
+
+  addGTAGFunctionFallback () {
+    const scriptContent = `function gtag() {
+      console.log('GoogleAnalytics: ' + [].slice.call(arguments));
+    }`
+    this.GTAGscriptElemFunctionFallback = document.createElement('script')
+    this.GTAGscriptElemFunctionFallback.type = 'text/javascript'
+    this.GTAGscriptElemFunctionFallback.text = scriptContent
+    document.head.appendChild(this.GTAGscriptElemFunctionFallback)
+  }
+
+  isValidId (gaId) {
+    if (gaId === 'debug') {
+      return true
+    } else {
+      return /^ua-\d{4,9}-\d{1,4}$/i.test(gaId)
+    }
+  }
+
+  checkOptState () {
+    if (typeof this.GTAGscriptElemFunctionFallback === 'undefined') {
+      this.addGTAGFunctionFallback()
+    }
+    if (typeof Cookies.get(this.cookieName) === 'undefined') {
+      this.showModal()
+    }
+    const getCookieValue = Cookies.get(this.cookieName) ? Cookies.get(this.cookieName) : 'false'
+    if (this.gaId) {
+      if (getCookieValue === 'false') {
+        Cookies.set(this.disableStr, true)
+        window[this.disableStr] = true
+      } else {
+        this.acceptTracking()
+        Cookies.remove(this.disableStr)
+        window[this.disableStr] = false
+      }
+    }
+  }
+
+  acceptTracking () {
+    Cookies.set(this.cookieName, true, { expires: Number(this.props.expiryDays), path: window.location.pathname })
+    this.hideModal()
+    if (this.gaId) {
+      window[this.disableStr] = false
+      Cookies.remove(this.disableStr)
+      if (this.props.serverSideTrackingEnabled === true) {
+        this.addGTAGScript()
+        this.addGTAGFunction()
+      }
+    }
+  }
+
+  declineTracking () {
+    Cookies.set(this.cookieName, false, { expires: Number(this.props.expiryDays), path: window.location.pathname })
+    this.hideModal()
+    if (this.gaId) {
+      Cookies.set(this.disableStr, true)
+      window[this.disableStr] = true
+      if (typeof this.GTAGscriptElem !== 'undefined') {
+        this.GTAGscriptElem.remove()
```

Comment:
GTAGscriptElem i would right the variable name like that
GtagScriptElement 
i think it is easier to read and in general we try to avoid abbreviations